### PR TITLE
Lany connection ingredient group

### PR DESCRIPTION
**Pull request by wlany**
I closed the pull request with only the addition of the GET request to get the ingredients of a group.
I merged that branch onto this branch where I connected group and ingredient. so this new pull request has the following additions:
- entities Group and Ingredient are connected (Group has `Set <Ingredient> ingredientsSet` and Ingredient has `Group group`).
- adapted the PUT /users/{userId}/ingredients call and tests accordingly (most changes in UserService method addUserIngredients)
- GET /groups/{groupId}/ingredients mapping (mostly as it was in the previous PR, but adapted the call in GroupController and the tests accordingly, but all will appear new here as it was never merged to development)

### CODE REVIEW

**Comment by akkuebler** [comment]
Tested it via Postman and compared with h2 -> works nicely! 
Only thought is the one conceirning the GET group/id/ingredients answer

File: src/main/java/ch/uzh/ifi/hase/soprafs23/controller/GroupController.java [hunk 1]
```diff
@@ -224,4 +227,30 @@ public List<UserGetDTO> getGroupMembersById(@PathVariable Long groupId, HttpServ
         return userGetDTOs;
     }
     
+    @GetMapping("/groups/{groupId}/ingredients")
```
[the remark this answers is not in the document]
Comment (wlany) [reply src/main/java/ch/uzh/ifi/hase/soprafs23/controller/GroupController.java:230]: I will add this to the REST table and add a dev task to corresponding US

File: src/main/java/ch/uzh/ifi/hase/soprafs23/rest/dto/IngredientGetDTO.java [hunk 2]
```diff
@@ -22,12 +21,4 @@ public void setName(String name) {
         this.name = name;
     }
 
-    public Long getCalculatedRating() {
```
Comment (akkuebler) [inline src/main/java/ch/uzh/ifi/hase/soprafs23/rest/dto/IngredientGetDTO.java]: thanks for the adjustments!

**Review by OrestisBollano — APPROVED** [review]
Everything seems ok :)

**Review by akkuebler — COMMENTED** [review]
nice

**Pull request merged**